app/src/main/java/com/rkm/tasky/network/repository/abstraction/TaskyRemoteRepository.kt

```diff
@@ -0,0 +1,46 @@
+package com.rkm.tasky.network.repository.abstraction
+
+import com.rkm.tasky.network.model.Reminder
+import com.rkm.tasky.network.model.Task
+import com.rkm.tasky.network.model.request.AccessTokenRequest
+import com.rkm.tasky.network.model.request.CreateEventRequest
+import com.rkm.tasky.network.model.request.LoginUserRequest
+import com.rkm.tasky.network.model.request.RegisterUserRequest
+import com.rkm.tasky.network.model.request.SyncAgendaRequest
+import com.rkm.tasky.network.model.request.UpdateEventRequest
+import com.rkm.tasky.network.model.response.AccessTokenResponse
+import com.rkm.tasky.network.model.response.AgendaResponse
+import com.rkm.tasky.network.model.response.AttendeeResponse
+import com.rkm.tasky.network.model.response.EventResponse
+import com.rkm.tasky.network.model.response.LoginUserResponse
+
+interface TaskyRemoteRepository {
+    suspend fun registerUser(request: RegisterUserRequest): Result<Unit>
+    suspend fun loginUser(request: LoginUserRequest): Result<LoginUserResponse>
+    suspend fun getNewAccessToken(request: AccessTokenRequest): Result<AccessTokenResponse>
+    suspend fun checkAuthentication(): Result<Unit>
+    suspend fun logoutUser(): Result<Unit>
```

Comment:
Could it make more sense to have these functions in a separate `interface`? This interface now demands too much of the class that needs to implement it. Also, performing user authentication related functions also does not seem like it should be apart of this interface.